waiter/src/waiter/util/utils.clj

```diff
@@ -93,13 +93,31 @@
   "Returns true if x is a non-negative integer"
   (and (integer? x) (non-neg? x)))
 
+(defn assoc-if-absent
+  "If the specified key, k, is not already associated with a value, v, in the map, m, associate k with v in m."
+  [m k v]
+  (cond-> m
+    (not (contains? m k)) (assoc k v)))
+
 (defn generate-secret-word
   [src-id dest-id processed-passwords]
   (let [password (second (first processed-passwords))
         secret-word (digest/md5 (str src-id ":" dest-id ":" password))]
     (log/debug "generate-secret-word" [src-id dest-id] "->" secret-word)
     secret-word))
 
+(let [server-name-atom (atom "waiter")]
```

Comment:
Why wrap this at all? Why not just have the bare atom? Less code, and I can do a 'find uses' of the atom and see all gets and sets, globally. Now, I'd need to search for the atom, then for these function uses.

Reply:
Almost everything in clojure is global, by design. What we lose by hiding it behind these accessors is 
#1: We have 9 lines of code; 10% of the lines in this patch and 33% of the new code added in this patch.
#2: We can no longer track accesses to the server name atom so easily, with, e.g., the IDE. We have to now trace indirectly via the finding the references to these two functions (versus searching for the one atom.)

The atom is still mutable, except instead of doing (reset! ...) its now (reset-server-name-atom!) and (get-current-server-name ...) instead of (deref).

Reply:
The flexibility is, currently, a hypothetical concern. But I would not be surprised if we needed to change the implementation in the future. E.g. in the case of messages chose to dynamically reload the messages from a properties file or some other store.

"Handful of accesses" is the current state, but may not be as the code evolves. 

I feel like this is a scenario ideal for the use of `mount`. We want to initialize some state (we generate from the settings) and then let it be freely read without worry of it being mutated. When we can control the mutation to only the initialization, there is no harm in having everyone else access it directly to read. I am mimicking that by restricting access to the raw atom and instead requiring invocation of a helper function to read the state.

I'd very much appreciate if you accept the patch, we can have a more general discussion on how these should be handled and perform a more complete refactoring on the codebase for such uses of atom once we reach a consensus.

Reply:
I'm fine with approving it, but apparently I don't have commit permissions to waiter